src/views/Nft/market/Profile/hooks/useFetchUserNfts.tsx

```diff
@@ -0,0 +1,27 @@
+import { useEffect } from 'react'
+import { useAppDispatch } from 'state'
+import { fetchUserNfts } from 'state/nftMarket/reducer'
+import { useUserNfts } from 'state/nftMarket/hooks'
+import usePreviousValue from 'hooks/usePreviousValue'
+import { useProfile } from 'state/profile/hooks'
+
+// We need to fetch collectibles for non-connected accounts, hence this hook accepts an account string.
+const useFetchUserNfts = (account: string) => {
+  const dispatch = useAppDispatch()
+  const { profile, isInitialized: isProfileInitialized } = useProfile()
+  const { isInitialized } = useUserNfts()
+  // TODO: This tokenId is showing for the previous user on account switch
+  const profileNftTokenId = profile?.tokenId?.toString()
+  const previousAccount = usePreviousValue(account)
+
+  const hasAccountSwitched = previousAccount !== account && isProfileInitialized
```

Comment:
`isProfileInitialized` seems to be unnecessary here. Both semantically and logically (shouldFetch checks for it and `shouldFetch` is checked later in useEffect)
Unless I'm missing some edge case 🤔

Reply:
Yea nice catch - these conditions did need refining, I've made some changes and added comments to hopefully help.